tests/unit/browser/test_cookies.py

```diff
@@ -0,0 +1,149 @@
+    with qtbot.waitSignal(ram_jar.changed, raising=True):
+        assert ram_jar.setCookiesFromUrl([cookie], url)
+
+    # assert the cookies are added correctly
+    all_cookies = ram_jar.cookiesForUrl(url)
+    assert len(all_cookies) == 1
+    saved_cookie = all_cookies[0]
+    assert (saved_cookie.name(),
+            saved_cookie.value()) == (cookie.name(), cookie.value())
+
+
+def test_set_cookies_never_accept(config_stub):
+    """Test setCookiesFromUrl when cookies are not accepted."""
+    config_stub.data = CONFIG_NEVER_COOKIES
+    ram_jar = cookies.RAMCookieJar()
+    error_signal_spy = QSignalSpy(ram_jar.changed)
+
+    assert not ram_jar.setCookiesFromUrl('test', 'test')
+    assert len(error_signal_spy) == 0
+
+
+def test_cookie_jar_init(config_stub, fake_save_manager):
+    """Test the CookieJar constructor."""
+    line_parser_stub = [cookie1, cookie2]
+    jar = cookies.CookieJar(line_parser=line_parser_stub)
+    assert objreg.get('save-manager').add_saveable.called
+
+    # Test that cookies are added to the jar
+    assert len(jar.allCookies()) == 2
+    assert jar.allCookies()[0].toRawForm().data() == cookie1
+    assert jar.allCookies()[1].toRawForm().data() == cookie2
+
+
+def test_purge_old_cookies(config_stub, fake_save_manager):
+    """Test that expired cookies are deleted."""
+    line_parser_stub = [cookie1, cookie2, session_cookie, expired_cookie]
+    jar = cookies.CookieJar(line_parser=line_parser_stub)
+
+    assert len(jar.allCookies()) == 4
+
+    jar.purge_old_cookies()
+
+    # Test that old cookies are gone
+    assert len(jar.allCookies()) == 3
+    assert jar.allCookies()[0].toRawForm().data() == cookie1
+    assert jar.allCookies()[1].toRawForm().data() == cookie2
+    assert jar.allCookies()[2].toRawForm().data() == session_cookie
+
+
+def test_save(config_stub, fake_save_manager, monkeypatch):
+    """Test that expired and session cookies are not saved."""
+    monkeypatch.setattr(lineparser,
+                        'LineParser', LineparserSaveStub)
+
+    jar = cookies.CookieJar()
+    jar._lineparser.data = [cookie1, cookie2, session_cookie, expired_cookie]
+
+    # Update the cookies on the jar itself
+    jar.parse_cookies()
+    jar.save()
+    assert len(jar._lineparser.saved) == 2
+    assert jar._lineparser.saved[0].data() == cookie1
+    assert jar._lineparser.saved[1].data() == cookie2
+
+
+def test_cookies_changed(config_stub, fake_save_manager, monkeypatch, qtbot):
+    """Test that self.changed is emitted and cookies are not saved."""
+    config_stub.data = CONFIG_COOKIES_ENABLED
+    monkeypatch.setattr(lineparser,
+                        'LineParser', LineparserSaveStub)
+    jar = cookies.CookieJar()
+    jar._lineparser.data = [cookie1, cookie2]
+
+    # Test that cookies are not saved
+    with qtbot.waitSignal(jar.changed, raising=True):
+        config_stub.set('content', 'cookies-store', False)
+
+    # Test that cookies are not saved
+    assert len(jar._lineparser.saved) == 0
```

Comment:
You should also test the reverse here, i.e. that cookies aren't deleted when it's set to `True`. Should probably be easy via `@pytest.mark.parametrize` (parametrize the config value and the expected length), and will give you 100% branch coverage too.